decode.go

```diff
@@ -269,6 +269,35 @@ func errTypeMismatch(dstType, srcType reflect.Type) *typeError {
 	return &typeError{dstType: dstType, srcType: srcType}
 }
 
+type unknownFieldError struct {
+	*errors.SyntaxError
```

Comment:
I don't like to change `syntaxError` to public interface .

For example, `syntaxError` can still be used as private interface if modified as follows .

```go
type unknownFieldError struct {
	err error
}

func (e *unknownFieldError) Error() string {
	return e.err.Error()
}

func errUnknownField(msg string, tk *token.Token) *unknownFieldError {
	return &unknownFieldError{err: errors.ErrSyntax(msg, tk)}
}
```